src/ReinforcementLearningEnvironments/src/environments/examples/CartPoleEnv.jl

```diff
@@ -19,88 +19,123 @@ Base.show(io::IO, params::CartPoleEnvParams) = print(
+    xthreshold = 2.4,
+)
+    CartPoleEnvParams{T}(
+        gravity,
+        masscart,
+        masspole,
+        masscart + masspole,
+        halflength,
+        masspole * halflength,
+        forcemag,
+        dt,
+        thetathreshold * π / 180,
+        xthreshold,
+        max_steps,
+    )
 end
 
+    mutable struct CartPoleEnv{A,T,ACT,R<:AbstractRNG} <: AbstractEnv
+        params::CartPoleEnvParams{T}
+        action_space::A
+        observation_space::Space{Vector{ClosedInterval{T}}}
+        state::Vector{T}
+        action::ACT
+        done::Bool
+        t::Int
+        rng::R
+    end
+
 """
     CartPoleEnv(;kwargs...)
 
 # Keyword arguments
 - `T = Float64`
+- `continuous = false`
+- `rng = Random.GLOBAL_RNG`
 - `gravity = T(9.8)`
 - `masscart = T(1.0)`
 - `masspole = T(0.1)`
 - `halflength = T(0.5)`
 - `forcemag = T(10.0)`
 - `max_steps = 200`
 - 'dt = 0.02'
-- `rng = Random.GLOBAL_RNG`
+- `thetathreshold = 12.0 # degrees`
+- `xthreshold` = 2.4``
 """
 function CartPoleEnv(;
     T = Float64,
-    gravity = 9.8,
-    masscart = 1.0,
-    masspole = 0.1,
-    halflength = 0.5,
-    forcemag = 10.0,
-    max_steps = 200,
-    dt = 0.02,
+    continuous = false,
     rng = Random.GLOBAL_RNG,
+    kwargs...,
 )
-    params = CartPoleEnvParams{T}(
-        gravity,
-        masscart,
-        masspole,
-        masscart + masspole,
-        halflength,
-        masspole * halflength,
-        forcemag,
-        dt,
-        2 * 12 * π / 360,
-        2.4,
-        max_steps,
+    params = CartPoleEnvParams(kwargs...)
+    action_space = continuous ? ClosedInterval{T}(-1.0, 1.0) : Base.OneTo(2)
+    state_space = Space(
+        ClosedInterval{T}[
+            (-2*params.xthreshold)..(2*params.xthreshold),
+            -1e38..1e38,
+            (-2*params.thetathreshold)..(2*params.thetathreshold),
+            -1e38..1e38,
+        ],
+    )
+    env = CartPoleEnv(
+        params, 
+        action_space,
+        state_space,
+        zeros(T, 4),
+        rand(action_space),
+        false,
+        0,
+        rng,
     )
-    high = cp = CartPoleEnv(params, zeros(T, 4), 2, false, 0, rng)
-    reset!(cp)
-    cp
+    reset!(env)
+    env
 end
 
-CartPoleEnv{T}(; kwargs...) where {T} = CartPoleEnv(; T = T, kwargs...)
+CartPoleEnv{T}(; kwargs...) where {T} = CartPoleEnv(T=T, kwargs...)
 
-function RLBase.reset!(env::CartPoleEnv{T}) where {T<:Number}
+Random.seed!(env::CartPoleEnv, seed) = Random.seed!(env.rng, seed)
+RLBase.action_space(env::CartPoleEnv) = env.action_space
+RLBase.state_space(env::CartPoleEnv) = env.observation_space
+RLBase.reward(env::CartPoleEnv{A,T}) where {A,T} = env.done ? zero(T) : one(T)
+RLBase.is_terminated(env::CartPoleEnv) = env.done
+RLBase.state(env::CartPoleEnv) = env.state
+
+function RLBase.reset!(env::CartPoleEnv{A,T}) where {A,T}
     env.state[:] = T(0.1) * rand(env.rng, T, 4) .- T(0.05)
     env.t = 0
-    env.action = 2
+    env.action = rand(env.action_space)
```

Comment:
```suggestion
    env.action = rand(env.rng, env.action_space)
```

Although this is not very important, it's better to not pollute the default rng here. 😃

Reply:
Makes sense. I was not really tracking the rng. I am sure this would have appeared after hours of trying to reproduce something.